Fixing problems that Jon reported (but introducing new ones by making the fixes) (#104)
mulesoft/docs-design-center: modules/ROOT/pages/design-named-examples.adoc

> = Guide to Named Examples in RAML 1.0

Reviewer: `Guide to defining examples in RAML 1.0`

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> If you want to use examples for a type, trait, or resource type in your RAML 1.0 API specification, you must follow these rules:

Reviewer: - `If you want to use examples...` -> `When defining examples...`
- we're talking about external/included examples here, so I would mention it -> `When defining examples in external files...`
- `must` is a little too prescriptive here IMO. I would either use `should` instead or rephrase a bit: `(...) , you can follow these simple rules:`

> * Use the `example` facet when you include the `.yaml` file for a single example.

Reviewer: `For a single example, you should use the `example` facet, and include a `.yaml` file`

> * Use the `examples` facet to include the `.raml` files for one or more NamedExample fragments.

Reviewer: ```
For multiple examples, you should always use the `examples` facet and either:
- define multiple examples, each pointing to individual `.yaml` files, with a single example per file; or
- include a `.raml` file containing one or more multiple examples defined in a single NamedExample fragment file
```

= Guide to Defining Examples in RAML 1.0
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> When defining examples in external files for a type, trait, or resource type in your RAML 1.0 API specification, you should follow these rules:

Author: @kbostrom , _sigh_... I wonder what their argument is. Something like this?

This use of "you" and the use of modal verbs are a type of weed in our topics.
We should devote ourselves to uprooting every single weed that appears in our documentation.
  Because uprooting weeds is a greater priority than pruning, watering, and otherwise tending to our strengths,
  And because every type weed is as pestilent as all of the other types.
Therefore, we should continue to divert ourselves from tending our strengths and instead deracinate this pestilent weed.

> * For a single example, you should use the `example` facet and include a `.yaml` file.

Author: @kbostrom , I've formatted the extension this way throughout the topics that I own, following the practice that I learned at IBM. I'll create a ticket to change it globally in the repos that I own, though.

As for your suggestion about replacing "you should use" with "use": I originally wrote "use", our RAML guru asked me to soften that to "you should use", as there is an alternative which API Designer should support, but does not, yet.

> * For multiple examples, you should always use the `examples` facet and either:

Author: Same comment as above. Our RAML guru requested this softening of the imperative because API Designer is not fully compliant with the RAML 1.0 spec.

** define multiple examples, each pointing to individual `.yaml` files, with a single example per file; or

> ** include a `.raml` file containing two or more examples defined in a single NamedExample fragment.

Reviewer 2: ```suggestion
** Include a `.raml` file containing two or more examples defined in a single NamedExample fragment
```

The reason for using `.yaml` files for single examples and not `.raml` files is that, according to the RAML 1.0 specification, a NamedExample fragment that defines one example is not valid. Therefore, you should use a NamedExample fragment only to define two or more examples.

For example, the following NamedExample fragments are invalid:

.fragment1.raml
----
#%RAML 1.0 NamedExample
givenName: ”Chiaki”
familyName: "Mukai"

> ----

Reviewer 2: Yay for international examples!

.fragment2.raml
----
#%RAML 1.0 NamedExample
givenName: "Kyung-won"
familyName: "Park"
----

> API Designer ignores the header and uses the examples. However, if you validate a specification with any other RAML processors, NamedExample fragments that define only one example should generate errors, in accord with the RAML 1.0 specification.

Reviewer 2: ```suggestion
API Designer ignores the header and uses the examples. However, if you validate a specification by using any other RAML processors, NamedExample fragments that define only one example generate errors, in accord with the RAML 1.0 specification.
```

Reviewer 2: Do we need "in accord..."? Does the audience need to know this?

Author: I have to use "should" here because we cannot definitively say what non-MuleSoft RAML processors do.

And yeah, we do need to state the "in accord" part. We're indirectly saying that API Designer is not fully compliant with RAML 1.0. We have to say this, but we don't want to put a sharp edge on it.

> == Using a single example that is in a `.yaml` file

Reviewer 2: ```suggestion
== Using a Single Example That Is in a `.yaml` File
```

> In the file `api.raml`, the `example` facet includes a single example that is in a `.yaml` file.

Reviewer 2: ```suggestion
In the file `api.raml`, the `example` facet includes a single example that is in a `.yaml` file:
```

    properties:
      givenName: string
      familyName: string
    example: !include fragment.yaml
----

.fragment.yaml
----
givenName: ”Chiaki”
familyName: "Mukai"
----

> == Naming two examples that are in `.yaml` files

Reviewer 2: ```suggestion
== Naming Two Examples That Are in `.yaml` Files
```

> In the file `api.raml`, the `examples` facet names the examples `fullName` and `otherFullName`. Each of the two examples is in a `.yaml` file.

Reviewer 2: ```suggestion
In the file `api.raml`, the `examples` facet names the examples `fullName` and `otherFullName`. Each of the two examples is in a `.yaml` file:
```

      givenName: string
      familyName: string
    examples:
      fullName: !include fragment1.yaml
      otherFullName: !include fragment2.yaml
----

.fragment1.yaml
----
givenName: ”Chiaki”
familyName: "Mukai"
----

.fragment2.yaml
----
givenName: "Kyung-won"
familyName: "Park"
----

> == Naming two examples that are in a NamedExample fragment

Reviewer 2: ```suggestion
== Naming Two Examples That Are in a NamedExample Fragment
```

> In the file `api.raml`, the `examples` facet does not name the examples.  The NamedExample fragment names, and it names them the examples `fullName` and `otherFullName`.

Reviewer 2: Not sure what is going on in the second sentence: missing word? Or did you mean to add that to the end of the first sentence.
